frontend/src/metabase/lib/urls/utils.ts

```diff
@@ -14,6 +14,17 @@ function flattenParam([key, value]: [string, unknown]) {
   return [[key, value]];
 }
 
-export function extractQueryParams(query: Record<string, unknown>) {
+function extractQueryParams(query: Record<string, unknown>) {
   return Object.entries(query).map(flattenParam).flat();
 }
+
+export function getEncodedUrlSearchParams(query: Record<string, unknown>) {
+  return new URLSearchParams(
+    extractQueryParams(query).map(([key, value]) => {
```

Comment:
This is pretty much the same logic from `frontend/src/metabase/lib/urls/questions.ts`, but I modify it a bit to make it read a bit easier.

Reply:
Awesome, looks much better. How about this though?
```
extractQueryParams(query).map(([key, value]) => [key, value == null ? "" : value])
```

Reply:
Thanks for the suggestion. Both are interchangeable to me. I still think having an if block makes it a bit more obvious what value we will return in each case, though it's a tad bit longer. And I also don't want to waste more cycles for CI to run too.